interp/cubic.go

```diff
@@ -297,3 +297,68 @@ func fritschButlandEdgeDerivative(xs, ys, slopes []float64, leftEdge bool) float
 	}
 	return g
 }
+
+// CubicSpline manage necessary equation for Natural cubic spline.
+type CubicSpline struct {
+	xs []float64
+
+	coeff *mat.Dense
+}
+
+// Fit fits a predictor to (X, Y) value pairs provided as two slices.
+func (cs *CubicSpline) Fit(xs, xy []float64) error {
+	n := len(xs)
+	cs.coeff = mat.NewDense((n-1)*4, 1, nil)
+	coeffs := mat.NewDense((n-1)*4, (n-1)*4, nil)
```

Comment:
I'm wondering if there is not a more efficient representation of this; `coeffs` is very sparse, with obvious structure in the data. @romanwerpachowski is there a banded representation that works for this? Outside my area, but brief reading says that this can be represented as a tridiagonal which would take us from O(n^2) space to O(n) space.

(Note that we don't currently implement GB solvers).

Reply:
Yes, it's a tridiagonal banded matrix. See e.g. http://www.inase.org/library/2014/athens/bypaper/MMCTSE/MMCTSE-01.pdf

Reply:
Yeah, that's what I thought, though the matrix that's created here is not tridiagonal, so I was unsure.

Reply:
I suspect that's because it's for the coefficients of a different polynomial (see one of my review comments).